src/org/labkey/test/aspects/ImpersonationLoggingAspect.java

```diff
@@ -43,7 +43,7 @@ public void beforeImpersonation(String impersonating)
             _impersonating = impersonating;
             _startTime = System.currentTimeMillis();
         }
-        else
+        else if (!_impersonating.equals(impersonating))
```

Comment:
Just curious why you have to do this check. Would there be any value in adding an else clause that basically says _impersonating equals impersonating so not going to do anything?

Reply:
After we pushed impersonation methods into the `UserMenu` component, we ended up having two methods in the call stack that matched this Aspect and they would both get logged, resulting in something like:
```
03:03:42,909   >>Impersonate - abstractor1ncitestproject@nlpui.test
03:03:42,910   ><Switch Impersonation : abstractor1ncitestproject@nlpui.test <1ms> -> abstractor1ncitestproject@nlpui.test
```